symplyphysics/laws/thermodynamics/equations_of_state/van_der_waals/dimensionless_equation.py

```diff
@@ -30,7 +33,33 @@
     Rational(8, 3) * reduced_temperature,
 )
 
-# TODO: derive from van der Waals equation of state
+# Derive from van der Waals equation of state
+
+_critical_pressure_expr = critical_pressure.law.rhs.subs({
+    critical_pressure.bonding_forces_parameter: equation.bonding_forces_parameter,
+    critical_pressure.molecules_volume_parameter: equation.molecules_volume_parameter,
+})
+
+_critical_molar_volume_expr = critical_molar_volume.law.rhs.subs({
+    critical_molar_volume.molecules_volume_parameter: equation.molecules_volume_parameter,
+})
+
+_critical_temperature_expr = critical_temperature.law.rhs.subs({
+    critical_temperature.bonding_forces_parameter: equation.bonding_forces_parameter,
+    critical_temperature.molecules_volume_parameter: equation.molecules_volume_parameter,
+})
+
+_reduced_eqn = equation.law.subs({
+    equation.pressure: reduced_pressure * _critical_pressure_expr,
```

Comment:
Ideally we should use a law to derive this. At least, point user that `p* = p/p_c` from the description above.

Reply:
I see there is some common pattern emerging. We have `molar` something, when we divide by amount of moles. We have `volumetric_density` something, when we divide by volume. We have `specific` something, when we divide by mass. And here we have `reduced` something, when we divide by critical value.

Reply:
Agree with all but `reduced`. It doesn't need to be a _critical_ value but can be any quantity of the same dimension as the original one that makes use of some characteristic parameters. E.g. [Lennard-Jones potential](https://en.wikipedia.org/wiki/Lennard-Jones_potential#Dimensionless_(reduced_units)).

Reply:
So with reduced values there will be separate definitions for each model that uses them. Like here, it's probably better to have three separate laws. But the `molar` and `specific` ones are all good.

Reply:
Well, it's up to you what you think looks better.